### PR TITLE
Update documentation of MatrixTools::apply_boundary_values().

### PR DESCRIPTION
**Pull request by bangerth**
Following the discussion on the mailing list.

### CODE REVIEW

File: include/deal.II/numerics/matrix_tools.h [hunk 1]
```diff
@@ -631,14 +631,22 @@ namespace MatrixCreator
...
  * <h3>Boundary conditions</h3>
  *
- * The apply_boundary_values() function inserts boundary conditions into a
- * system of equations.  To actually do this you have to specify a list of
- * degree of freedom indices along with the values these degrees of freedom
- * shall assume. To see how to get such a list, see the discussion of the
- * VectorTools::interpolate_boundary_values function.
+ * The apply_boundary_values() functions modifies a linear system to incorporate
+ * the constraints that result from Dirichlet-type boundary conditions (or, more
+ * specifically: "strong" boundary condition). To actually do this, the
```
Comment (kronbichler) [inline include/deal.II/numerics/matrix_tools.h:636]:
```suggestion
 * specifically: "strong" boundary conditions). To actually do this, the
```

File: include/deal.II/numerics/matrix_tools.h [hunk 2]
```diff
@@ -631,14 +631,22 @@ namespace MatrixCreator
...
+ * freedom indices along with the values these degrees of freedom should have.
+ * To see how to get such a list, see the discussion of the
+ * VectorTools::interpolate_boundary_values() function as one example.
  *
  * There are two ways to incorporate fixed degrees of freedom such as boundary
- * nodes into a linear system, as discussed below.
+ * nodes into a linear system, as discussed below. Both operate at either the
+ * level of local contributions to the global linear system, or the global
+ * system itself. A third way, using the
+ * ConstraintMatrix::copy_local_to_global() performs the same process as part of
```
Comment (kronbichler) [inline include/deal.II/numerics/matrix_tools.h:646]:
```suggestion
 * AffineConstraints::distribute_local_to_global() performs the same process as part of
```

**Comment by bangerth** [comment]
Thanks, @kronbichler . Adjusted as requested!

**Comment by bangerth** [comment]
Well, actually no -- the airport wireless allows for web traffic but not my git connection. Will get back to you once it's updated.

**Comment by bangerth** [comment]
OK, now for real!

File: include/deal.II/numerics/matrix_tools.h [hunk 3]
```diff
@@ -631,14 +631,22 @@ namespace MatrixCreator
...
+ * functions of this name in the current namespace require a list of degree of
+ * freedom indices along with the values these degrees of freedom should have.
+ * To see how to get such a list, see the discussion of the
+ * VectorTools::interpolate_boundary_values() function as one example.
  *
  * There are two ways to incorporate fixed degrees of freedom such as boundary
- * nodes into a linear system, as discussed below.
+ * nodes into a linear system, as discussed below. Both operate at either the
+ * level of local contributions to the global linear system, or the global
+ * system itself. A third way, using the
```
Comment (drwells) [inline include/deal.II/numerics/matrix_tools.h:645]:
```suggestion
 * system itself. A third way, using
```

File: include/deal.II/numerics/matrix_tools.h [hunk 4]
```diff
@@ -631,14 +631,22 @@ namespace MatrixCreator
...
+ * AffineConstraints::copy_local_to_global() performs the same process as part
```
Comment (drwells) [inline include/deal.II/numerics/matrix_tools.h:646]:
```suggestion
 * AffineConstraints::copy_local_to_global(), performs the same process as part
```

File: include/deal.II/numerics/matrix_tools.h [hunk 5]
```diff
@@ -838,8 +854,8 @@ namespace MatrixTools
 
   /**
    * Apply Dirichlet boundary conditions to the system matrix and vectors as
-   * described in the general documentation. This function works for block
-   * sparse matrices and block vectors
+   * described in the general documentation of this namespace. This function
+   * works for block sparse matrices and block vectors
```
Comment (drwells) [inline include/deal.II/numerics/matrix_tools.h:858]:
```suggestion
   * works for block sparse matrices and block vectors.
```

**Review by drwells — CHANGES_REQUESTED** [review]
I have a few minor suggestions but overall this looks good.

**Comment by bangerth** [comment]
@drwells Updated as requested. Thanks for the careful reading!

**Comment by bangerth** [comment]
@drwells Updated as requested. Thanks for the careful reading!

**Review by drwells — APPROVED** [review]

**Pull request merged**